### PR TITLE
Upgrade dependencies to use Gatsby V2

### PR DESCRIPTION
**Pull request by glauberfc**
I upgraded the dependencies of project and when I run `gatsby develop` the console returns this error:

```
success open and validate gatsby-configs — 0.011 s
error value must be an array of bytes

TypeError: value must be an array of bytes

  - v35.js:29 generateUUID
    [gatsby-starter-docs]/[uuid]/lib/v35.js:29:38

  - create-node-id.js:16 createNodeId
    [gatsby-starter-docs]/[gatsby]/dist/utils/create-node-id.js:16:10

  - load.js:131 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:131:13

  - load.js:110 plugin.options.plugins.forEach.p
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:110:27

  - Array.forEach

  - load.js:109 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:109:32

  - load.js:148 config.plugins.forEach.plugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:148:20

  - Array.forEach

  - load.js:147 module.exports
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:147:20

  - index.js:56
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/index.js:56:21

  - Generator.next

  - debuggability.js:313 Promise._execute
    [gatsby-starter-docs]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [gatsby-starter-docs]/[bluebird]/js/release/promise.js:483:18
error value must be an array of bytes


TypeError: value must be an array of bytes

  - v35.js:29 generateUUID
    [gatsby-starter-docs]/[uuid]/lib/v35.js:29:38

  - create-node-id.js:16 createNodeId
    [gatsby-starter-docs]/[gatsby]/dist/utils/create-node-id.js:16:10

  - load.js:131 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:131:13

  - load.js:110 plugin.options.plugins.forEach.p
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:110:27

  - Array.forEach

  - load.js:109 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:109:32

  - load.js:148 config.plugins.forEach.plugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:148:20

  - Array.forEach

  - load.js:147 module.exports
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:147:20

  - index.js:56
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/index.js:56:21

  - Generator.next

  - debuggability.js:313 Promise._execute
    [gatsby-starter-docs]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [gatsby-starter-docs]/[bluebird]/js/release/promise.js:483:18

error UNHANDLED REJECTION

 TypeError: value must be an array of bytes

  - v35.js:29 generateUUID
    [gatsby-starter-docs]/[uuid]/lib/v35.js:29:38

  - create-node-id.js:16 createNodeId
    [gatsby-starter-docs]/[gatsby]/dist/utils/create-node-id.js:16:10

  - load.js:131 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:131:13

  - load.js:110 plugin.options.plugins.forEach.p
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:110:27

  - Array.forEach

  - load.js:109 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:109:32

  - load.js:148 config.plugins.forEach.plugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:148:20

  - Array.forEach

  - load.js:147 module.exports
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:147:20

  - index.js:56
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/index.js:56:21

  - Generator.next

  - debuggability.js:313 Promise._execute
    [gatsby-starter-docs]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [gatsby-starter-docs]/[bluebird]/js/release/promise.js:483:18

error Command failed with exit code 1.

error UNHANDLED REJECTION

TypeError: value must be an array of bytes

  - v35.js:29 generateUUID
    [gatsby-starter-docs]/[uuid]/lib/v35.js:29:38

  - create-node-id.js:16 createNodeId
    [gatsby-starter-docs]/[gatsby]/dist/utils/create-node-id.js:16:10

  - load.js:131 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:131:13

  - load.js:110 plugin.options.plugins.forEach.p
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:110:27

  - Array.forEach

  - load.js:109 processPlugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:109:32

  - load.js:148 config.plugins.forEach.plugin
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:148:20

  - Array.forEach

  - load.js:147 module.exports
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/load.js:147:20

  - index.js:56
    [gatsby-starter-docs]/[gatsby]/dist/bootstrap/load-plugins/index.js:56:21

  - Generator.next

  - debuggability.js:313 Promise._execute
    [gatsby-starter-docs]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [gatsby-starter-docs]/[bluebird]/js/release/promise.js:483:18

error Command failed with exit code 1.
```

### CODE REVIEW

**Comment by levino** [comment]
Please do not share screenshots of log files. Share the logs as text (copy and paste)

**Comment by levino** [comment]
I think you did not follow[ the guide on how to migrate from gatsby v1 to v2](https://www.gatsbyjs.org/docs/migrating-from-v1-to-v2/), did you? I suggest you work through this guide and then we continue this conversation. Just updating the dependencies is far from enough.

**Comment by glauberfc** [comment]
Now I finished the guide of migration. May I make a new push now?

**Comment by levino** [comment]
Yes please push whatever you have

**Comment by glauberfc** [comment]
The same errors are returned

**Comment by levino** [comment]
Looks like there was a bug with prettier. Your application of prettier to the root .js files created a larger diff than necessary. Please wait for #45 to be merged, then rebase onto master and update this PR to make the diff smaller.

File: src/components/Layout/index.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+//       title = `Tagged in ${capitalize(tag)}`
+//     } else if (currentPath.includes('categories/')) {
+//       const category = currentPath
+//         .replace('categories/', '')
+//         .replace('/', '')
+//         .replace('-', ' ')
+//       title = `${capitalize(category)}`
+//     }
+//     return title
+//   }
```
Comment (levino) [inline src/components/Layout/index.jsx]: Why is this code commented out? Please either remove it completely or leave it in. What is the background on this?

File: src/components/Layout/index.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// export ({ children }) => (
+//   <div>
+//     <Helmet>
+//       <title>{`${config.siteTitle}`}</title>
+//       <meta name="description" content={config.siteDescription} />
+//     </Helmet>
+//     <ThemeProvider theme={theme}>{children}</ThemeProvider>
+//   </div>
+// )
```
Comment (levino) [inline src/components/Layout/index.jsx]: Same as above.

**Comment by levino** [comment]
It is nice to share logs in a formatted way like so:

```bash
$ git commit --amend
[feature/prettier b184e85] Apply prettier to all .js and .jsx files in the codebase.
 Date: Fri Nov 2 08:05:56 2018 +0100
 5 files changed, 4048 insertions(+), 4423 deletions(-)
 rewrite data/SiteConfig.js (75%)
```
You might want to read about [github flavored markdown](https://github.github.com/gfm/).

**Comment by levino** [comment]
I have to work now. Will checkout and play around this afternoon.

File: src/components/Layout/index.jsx (hunk 1 above)
Comment (glauberfc) [reply src/components/Layout/index.jsx]: I'm sorry. I hadn't certain about this changes.

**Comment by levino** [comment]
[I fixed all the bugs preventing building](https://github.com/glauberfc/gatsby-starter-docs/pull/1). Please merge it in @glauberfc by merging the PR.

Building works, development preview works. So now the question is: Did anything break? Once @glauberfc merges my PR we should get a working preview on netlify, then we can see what has broken.

I think in this codebase is a lot of unnecessary code which is very bad for an example for people "getting started" with gatsby: They will get confused. Someone should throw out every line of code that is not used atm.

**Comment by levino** [comment]
@ericwindmill Can you please check out whether there is anything broken and report back?

**Comment by marvindanig** [comment]
> @ericwindmill Can you please check out whether there is anything broken and report back?

Did this go anywhere?

**Pull request closed without merging**